content/en/platform/corda/5.0-beta/corda-cli-reference/cli-commands/rpc-permission-templates.md

```diff
@@ -0,0 +1,98 @@
+---
+date: '2023-01-10'
+menu:
+  corda-5-beta:
+    identifier: corda-5-beta-rpc-permissions-templates
+    weight: 6000
+    parent: corda-5-cli-reference
+section_menu: corda-5-beta
+title: "RPC permission templates"
+---
+
+## Create Common RPC Permission Templates
+
+Currently, when the C5 Cluster starts there is a single "admin" account auto-created which can do everything. While this account can be used to perform any action, there is room for error if not used carefully. The idea is to create fine-grained roles for specific actions such as:
+
+• A dedicated role which can create users, roles, permissions and drive all the associations between them;
+• A dedicated role with a set of all the necessary permissions to create a virtual node (including CPI upload);
+• A dedicated role which allows flows to run on this vNode.
+
+{{< note >}}
+These roles and permissions enable certain common operations. Creation of users and associating such users to the roles should be still done manually by an admin user.
+{{< /note >}}
+
+UserAdminRole- Creates roles, permissions, and controls all associations between the user, roles, and permissions. Create this role at cluster bootstrap by the admin user. There is no direct write into DB like previously done with the admin user.
+
+VNodeCreatorRole- Set all the necessary permissions to create a virtual node, including CPI upload and CPI update. Create this role at cluster bootstrapping time.
+
+CordaDeveloperRole- Users can do the following:
+
+1. VNode reset
+2. Vnode status update
+
+This role should be provisioned at cluster boostrap time and should re-use previously created permissions for other roles.
+
+FlowExecutorRole- Once the vNode is created, a new CLI command will permit creation of such role for a given vNode.
+
+## Ability to query permissions via HTTP RPC
+
+At the moment it is only possible to retrieve a permission if its identifier is known to the caller.
+However, there might be cases when a new role is being setup and existing permissions need to be included. It would be helpful to do a search of the permissions available.
+
+{{< note >}}
+Being unable to do this creates potentially duplicated permissions.
+{{< /note >}}
+
+## Implement Permissions Check When Starting a Particular Flow
+
+Currently, the check is made to assert whether a user can execute startFlow RPC operations. No checks are made to whether the user can start a particular flow.
+
+These checks should be performed against the RBAC sub-system even before passing start request to FlowWorker.
+
+## RPC Permission Templates: Test Plan Draft
```

Comment:
I wonder if we need to include "Test Plan Draft" into potentially end-user facing document?
Feels like it is more of an internal script which our QA team been following.